drivers/flash/spi_flash_w25qxxdv.c

```diff
@@ -367,8 +367,8 @@ static int spi_flash_init(struct device *dev)
 	SYNC_INIT();
 
 	ret = spi_flash_wb_configure(dev);
-	if (!ret) {
-		dev->driver_api = &spi_flash_api;
+	if (ret) {
+		return ret;
```

Comment:
Probably pedantry, but this changes the logic: before, if the configure() call failed, the rest of the function would have been executed (which does some stuff related to "page layout").  Now that gets skipped.  That's probably OK, but given that this code that doesn't get tested under CI, it would be prudent to leave the logic unchanged for this kind of refactoring.  Or to make sure it gets tested (i.e. test the configure failure case!) on appropriate hardware, or at least get buy-in from someone responsible for the hardware.

Reply:
ok, so updated code, now returning in end only. this will not change current logic & if error is returned by spi_flash_wb_configure(),  init() function will return non zero value and driver_api will be set to NULL as per commit a68120de6d414117ddb67566a7397a5e23556841